src/darker/formatters/formatter_config.py

```diff
@@ -0,0 +1,73 @@
+"""Code re-formatter plugin configuration type definitions."""
+
+from __future__ import annotations
+
+import re
+from decimal import Decimal
+from typing import TYPE_CHECKING, Iterable, Pattern, TypedDict
+
+from darkgraylib.config import ConfigurationError
+
+if TYPE_CHECKING:
+    from argparse import Namespace
+
+
+class FormatterConfig(TypedDict):
+    """Base class for code re-formatter configuration."""
```

Comment:
If this is the only place this is used can probably jsut remove it

Reply:
The whole `FormatterConfig` class?

I'm using it for type checking. All configuration classes for different formatters must subclass this one, and Mypy can then check the types correctly.

I did consider using [class protocols] instead of subclassing. But this is such a simple class hierarchy that I'm not sure if it pays off.

[class protocols]: https://docs.python.org/3/library/typing.html#typing.Protocol